### PR TITLE
Implementation of ClusterPairing and Migration API's for the SDK.

### PR DESCRIPTION
**Pull request by sharma-tapas**
**What this PR does / why we need it**:
This commit contains the API's in the SDK for the ClusterPairing and VolumeMigration use cases.
Signed-off-by: Tapas Sharma <tapas@portworx.com>

**Which issue(s) this PR fixes** (optional)
Part of #653

**Special notes for your reviewer**:
@lpabon This is a cleaner PR with all commits squashed into one and lesser merge conflicts.

### CODE REVIEW

**Comment by lpabon** [comment]
This looks like a replacement for #692, right? We need to make sure @disrani-px is ok with the review here also.

File: api/server/sdk/server.go [hunk 1]
```diff
@@ -369,6 +372,32 @@ func (s *Server) restServerSetupHandlers() (*http.ServeMux, error) {
...
+	err = api.RegisterOpenStorageMigrateHandlerFromEndpoint(
+		context.Background(),
+		gmux,
+		s.Address(),
+		[]grpc.DialOption{grpc.WithInsecure()})
+	if err != nil {
+		return nil, err
+	}
+
+	err = api.RegisterOpenStorageMigrateHandlerFromEndpoint(
```
Comment (lpabon) [inline api/server/sdk/server.go]: I think this one is not needed since 384 is adding it already.

**Comment by lpabon** [comment]
Also, from #692 @disrani-px and I discussed it and here are the results:

* [Token](https://github.com/libopenstorage/openstorage/pull/692#discussion_r230204079): Use GetToken, since golang protobuf does this anyways for all accesses.
* [Checking parameters](https://github.com/libopenstorage/openstorage/pull/692#discussion_r230206386): @disrani-px will review this PR and let you know which of the values not to check for. We will also be adding an issue to have the inner/Golang interfaces the ability to return known errors so that the SDK does not just return codes.Internal.
* [ResetToken](https://github.com/libopenstorage/openstorage/pull/692#discussion_r229874517): Using ResetToken is ok
* [Idempotency](https://github.com/libopenstorage/openstorage/pull/692#discussion_r230209700): `name` will be changed back to `task_id` and be made optional. This will allow the client to determine if they want the function to be idempotent or not. The documentation for `task_id` will show to set the value for idempotency.

File: api/server/sdk/server.go (hunk 1 above)
Comment (sharma-tapas) [reply api/server/sdk/server.go]: Done.

**Comment by sharma-tapas** [comment]
@lpabon  Following comments were addressed in this commit,
* Token: Used GetToken.
* Idempotency: name is changed back to task_id and is made optional.

Waiting for @disrani-px  for the following,
* Checking parameters: @disrani-px will review this PR and let you know which of the values not to check for. We will also be adding an issue to have the inner/Golang interfaces the ability to return known errors so that the SDK does not just return codes.Internal.

**Pull request merged**